vmhooks/baseOps.md

```diff
@@ -103,6 +108,13 @@ module BASEOPS
       requires IDX <Int size(ARGS)
        andBool isBytes(ARGS[IDX])
 
+    rule <instrs> hostCall("env", "getArgumentLength", [ i32 .ValTypes ] -> [ i32 .ValTypes ])
+               => #throwException(ExecutionFailed, "invalid argument") ... 
+         </instrs>
+         <locals> 0 |-> <i32> IDX </locals>
+         <callArgs> ARGS </callArgs>
+      requires IDX >=Int size(ARGS)
```

Comment:
Why this rule and the one above use IDX as an unsigned, while the two rules below use it as a signed value?

In particular, why the two rules above do not test how `#signed(i32, IDX)` compare with 0?

Reply:
That's right. Now I added negativity checks to these rules in [this commit](https://github.com/runtimeverification/elrond-semantics/pull/97/commits/173977c7e5cb31a83f62f654386aedf5bded0032).